ra/ra.go

```diff
@@ -1289,13 +1289,14 @@ func (ra *RegistrationAuthorityImpl) enforceNameCounts(ctx context.Context, name
 
 	response, err := ra.SA.CountCertificatesByNames(ctx, req)
 	if err != nil {
-		return nil, err
+		return nil, time.Time{}, err
 	}
 
 	if len(response.Counts) == 0 {
-		return nil, errIncompleteGRPCResponse
+		return nil, time.Time{}, errIncompleteGRPCResponse
 	}
 
+	var earliest = time.Unix(0, response.Earliest)
```

Comment:
If an RA gets this change before the SA does, then the RA will believe that the SA's (incomplete) response was actually encoding a value of `0` for `earliest`. This should have a check to only use the value of `earliest` if it is non-zero; or the PR should be split into two halves, with the server half being merged and deployed first.

Reply:
I don't believe this to be the case. Above, on line `1295`, the RA treats responses from the SA with 0 counts as an incomplete response from the SA and returns an `errIncompleteGRPCResponse` to the caller of `.enforceNameCounts()`.

Reply:
Ah, scratch that, I see what you mean. The call is coming from inside the RA, of course.

Reply:
Oh, actually, no I think this still works. The caller is ra.checkCertificatesPerNameLimit(), which will return immediately after receiving that `errIncompleteGRPCResponse`.